src/Microsoft.SqlTools.ServiceLayer/Rename/RenameService.cs

```diff
@@ -0,0 +1,138 @@
+        internal static ConnectionService ConnectionServiceInstance
+        {
+            get
+            {
+                if (connectionService == null)
+                {
+                    connectionService = ConnectionService.Instance;
+                }
+                return connectionService;
+            }
+            set
+            {
+                connectionService = value;
+            }
+        }
+
+        public void InitializeService(IProtocolEndpoint serviceHost)
+        {
+            this.serviceHost = serviceHost;
+            this.serviceHost.SetRequestHandler(ProcessRenameEditRequest.Type, HandleProcessRenameEditRequest);
+
+        }
+        internal Task HandleRequest<T>(RequestContext<T> requestContext, Func<Task> action)
+        {
+            // The request handling will take some time to return, we need to use a separate task to run the request handler so that it won't block the main thread.
+            // For any specific table designer instance, ADS UI can make sure there are at most one request being processed at any given time, so we don't have to worry about race conditions.
+            return Task.Run(async () =>
+            {
+                try
+                {
+                    await action();
+                }
+                catch (Exception e)
+                {
+                    await requestContext.SendError(e);
+                }
+            });
+        }
+        /// <summary>
+        /// Method to handle the renaming operation
+        /// </summary>
+        /// <param name="requestParams">parameters which are needed to execute renaming operation</param>
+        /// <param name="requestContext">Request Context</param>
+        /// <returns></returns>
+        internal Task HandleProcessRenameEditRequest(ProcessRenameEditRequestParams requestParams, RequestContext<bool> requestContext)
+        {
+            return this.HandleRequest<bool>(requestContext, async () =>
+           {
+               Logger.Verbose("Handle Request in ProcessRenameEditRequest()");
+               bool operationExecutedSuccessFull = false;
+               RenameUtils.Validate(requestParams);
+               ConnectionInfo connInfo;
+               try
+               {
+                   connectionService.TryFindConnection(
+                          requestParams.TableInfo.OwnerUri,
+                          out connInfo);
+
+                   using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, "RenamingDatabaseObjects"))
+                   {
+                       ExecuteRenaming(requestParams, sqlConn);
+                   }
+                   operationExecutedSuccessFull = true;
+               }
+               catch (Exception e)
+               {
+                   Logger.Error("Error on executing renaming operation: " + e.ToString());
+                   throw new InvalidOperationException("The renaming operation was not successfull executed");
```

Comment:
shouldn't the exception message be a localizable resource?

Reply:
That's absolutely true. I will add it.

Reply:
not sure why you picked InvalidOperationException type, that doesn't seem to be the right type, I don't mind if you use Exception directly.

Reply:
I suggest that each component define its own exception type so the source of errors is very clear to developers and eventually to customer support.
Also it becomes very hard to debug something if you have to set your debugger exception filter to catch ALL first chance exceptions because your component throws "Exception". 
Choosing the scope for "Component" is perhaps interesting. If the tools service has already defined any common exception type it could be used here. Otherwise just creating a new type for this specific service and throw it, wrapping the handled exception as the InnerException.